jobserver/views/users.py

```diff
@@ -108,4 +112,10 @@ def get_queryset(self):
             raise_if_not_int(backend)
             qs = qs.filter(backends__pk=backend)
 
+        role = self.request.GET.get("role")
+        if role:
+            qs = qs.alias(extracted=ExtractRoles("roles")).filter(
```

Comment:
nit (and fine to leave for now) but structurally I wonder if this logic belongs as a method on a custom QuerySet on the `User` model. Something like `filter_by_role`? I can imagine this is something we might want to do in various places and it's really the models responsibility to worry about the implementation details here.

Reply:
I had a similar thought and opted to leave it here until it's used elsewhere.  I could see it going either way but I'm reasonably certain there's no short or medium term plans to need it outside this particular view, so my reasoning was adding an abstraction only used in one place wouldn't be overly useful.

Reply:
Got you. Makes sense.

Reply:
Having looked at your comment on the quoting of role names, I am entirely changing my mind and agreeing with you, go team 😁